src/sqlfluff/core/parser/segments/base.py

```diff
@@ -84,11 +84,13 @@ class BaseSegment:
     allow_empty = False
     # What other kwargs need to be copied when applying fixes.
     additional_kwargs: List[str] = []
+    # This typing isnt really true as we set the pos_marker to None in places
+    pos_marker: PositionMarker
```

Comment:
I took a look at this on my machine. There's definitely a lot of places that assume `pos_marker` is set, even though it may not be. It's probably best to leave it as you have it.